src/app/_components/BlogSearchInput.tsx

```diff
@@ -2,30 +2,48 @@
 
 import { useState } from 'react'
 
+import { useSearchParams } from 'next/navigation'
+
 import { useDebounceCallback } from 'usehooks-ts'
 
+import { useSearch } from '@/hooks/useSearch'
 import { useUpdateSearchParams } from '@/hooks/useUpdateSearchParams'
 
 import { SearchInput } from '@/components/SearchInput'
 
+import { PAGE_KEY, SEARCH_KEY } from '@/constants/searchParams'
+
 type BlogSearchInputProps = {
-  searchQuery: string
+  query: ReturnType<typeof useSearch>['searchQuery']
 }
 
-export function BlogSearchInput({ searchQuery }: BlogSearchInputProps) {
-  const [value, setValue] = useState(searchQuery)
+const TYPE_DEBOUNCE_DELAY = 400
+
+export function BlogSearchInput({ query }: BlogSearchInputProps) {
+  const [value, setValue] = useState(query)
+  const params = useSearchParams()
 
   const updateSearchParams = useUpdateSearchParams()
 
   const debouncedUpdateSearchParams = useDebounceCallback(
     updateSearchParams,
-    500,
+    TYPE_DEBOUNCE_DELAY,
   )
 
   function handleSearchChange(event: React.ChangeEvent<HTMLInputElement>) {
     const newValue = event.target.value
     setValue(newValue)
-    debouncedUpdateSearchParams({ search: newValue, page: '1' })
+
+    const existingPageParam = params.get(PAGE_KEY)
+
+    if (existingPageParam && existingPageParam !== '1') {
```

Comment:
What is `1` the default, the first page?

Reply:
Ah yes! `1` is the first page. Basically, we want to reset the pagination to the first page when users search to avoid weird behaviours.

Maybe we can `const FIRST_PAGE = 1`

Reply:
Yeah - We have `DEFAULT_PAGE_NUMBER` in `validatePageNumber` - Could that be used?

This is what I did for `sortConstants`

```
import { type SortOptionType, type SortSetting } from '@/types/sortTypes'

export const DEFAULT_SORT_OPTION: SortOptionType = 'newest'

export const sortSettings: SortSetting[] = [
  { id: 'newest', name: 'Newest' },
  { id: 'oldest', name: 'Oldest' },
] as const

export const VALID_SORT_OPTIONS = sortSettings.map((setting) => setting.id)
```

I know it's unlikely we will ever change the default page to anything but `1` - and yet, it feels good to have this in one place - What do you think?

Reply:
Yes let's use the existing `DEFAULT_PAGE_NUMBER`. We will never change it to anything other than "1" but it reads better and feels cleaner :)